### PR TITLE
40percentclub sixpack support

### PR DESCRIPTION
**Pull request by Suschman**
Di0ib sixpack support

## Description
- Add support for 40percentclub sixpack board
- Use VIA style vid and pid
- Implements backlight and breathing (as original firmware breathed as well)
- Has default layout as per default configuration in original firmware
- qmk cformat used

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/40percentclub/sixpack/rules.mk
Comment (zvecr) [inline keyboards/40percentclub/sixpack/rules.mk:33]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/40percentclub/sixpack/sixpack.h
Comment (zvecr) [inline keyboards/40percentclub/sixpack/sixpack.h:31]:
```suggestion
#define LAYOUT( \
    k00, k01, k02, \
    k10, k11, k12  \
) { \
    { k00, k01, k02 }, \
    { k10, k11, k12 }  \
}
```

File: keyboards/40percentclub/sixpack/sixpack.c
Comment (zvecr) [inline keyboards/40percentclub/sixpack/sixpack.c:55]:
```suggestion

void matrix_init_kb(void) {
    setPinOutput(B6);  // Backlight cathodes Col.3
    setPinOutput(F6);  // Backlight cathodes Col.2
    setPinOutput(F7);  // Backlight cathodes Col.1

    matrix_init_user();
}
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by Suschman** [comment]
Thanks for the review.
Would like to add VIA support and an additional keymap, can this be included with this PR, or better separate after merge?

**Review by drashna — APPROVED** [review]

**Comment by fauxpark** [comment]
@Suschman yes, you can include that in this PR :)

File: keyboards/40percentclub/sixpack/readme.md
Comment (fauxpark) [inline keyboards/40percentclub/sixpack/readme.md:15]:
```suggestion
    make 40percentclub/sixpack:default

Flashing example for this keyboard:

    make 40percentclub/sixpack:default:flash
```

File: keyboards/40percentclub/sixpack/keymaps/via/rules.mk
Comment (fauxpark) [inline keyboards/40percentclub/sixpack/keymaps/via/rules.mk:5]:
```suggestion
VIA_ENABLE = yes
```

File: keyboards/40percentclub/sixpack/keymaps/via/keymap.c
Comment (fauxpark) [inline keyboards/40percentclub/sixpack/keymaps/via/keymap.c:43]:
```suggestion
```

File: keyboards/40percentclub/sixpack/keymaps/fkeys/rules.mk
Comment (fauxpark) [inline keyboards/40percentclub/sixpack/keymaps/fkeys/rules.mk:5]:
```suggestion
TAP_DANCE_ENABLE = yes
```

File: keyboards/40percentclub/sixpack/keymaps/fkeys/keymap.c
Comment (fauxpark) [inline keyboards/40percentclub/sixpack/keymaps/fkeys/keymap.c:52]:
```suggestion
```

File: keyboards/40percentclub/sixpack/keymaps/fkeys/config.h
Comment (fauxpark) [inline keyboards/40percentclub/sixpack/keymaps/fkeys/config.h:1]: This file can be deleted.

File: keyboards/40percentclub/sixpack/config.h
Comment (fauxpark) [inline keyboards/40percentclub/sixpack/config.h:63]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/40percentclub/sixpack/keymaps/fkeys/config.h
Comment (Suschman) [reply keyboards/40percentclub/sixpack/keymaps/fkeys/config.h]: removed

**Review by fauxpark — APPROVED** [review]

File: keyboards/40percentclub/sixpack/config.h
Comment (drashna) [inline keyboards/40percentclub/sixpack/config.h:23]: This VID is not in use, but it is very close to what USB IF is issuing for new IDs.   I would prefer that this was something higher, to prevent collisions in the near future.

**Review by drashna — APPROVED** [review]

File: keyboards/40percentclub/sixpack/config.h
Comment (Suschman) [reply keyboards/40percentclub/sixpack/config.h:23]: VID was selected to align with the other 40percentclub boards using it (or FEED/CEEB), and per VIA recommendations.

`25/config.h:#define VENDOR_ID       0xFEED`
`4pack/config.h:#define VENDOR_ID 0xFEED`
`4x4/config.h:#define VENDOR_ID       0xFEED`
`5x5/config.h:#define VENDOR_ID       0xFEED`
`6lit/config.h:#define VENDOR_ID       0xFEED`
`foobar/config.h:#define VENDOR_ID       0xFEED`
`gherkin/config.h:#define VENDOR_ID       0xFEED`  ([#11040](https://github.com/qmk/qmk_firmware/pull/11040#issuecomment-735672452))
`half_n_half/config.h:#define VENDOR_ID       0xFEED`
`i75/config.h:#define VENDOR_ID       0xFEED`
`luddite/config.h:#define VENDOR_ID       0x3430 // "40"`
`mf68/config.h:#define VENDOR_ID       0xCEEB`
`nano/config.h:#define VENDOR_ID       0xCEEB`
`nein/config.h:#define VENDOR_ID       0x3430 // "40"`
`nori/config.h:#define VENDOR_ID       0xFEED`
`sixpack/config.h:#define VENDOR_ID       0x3430 // "40"`
`tomato/config.h:#define VENDOR_ID       0xFEED`
`ut47/config.h:#define VENDOR_ID       0xFEED`

Comment (fauxpark) [reply keyboards/40percentclub/sixpack/config.h:23]: The VIA docs specifically recommend changing the VID: https://caniusevia.com/docs/configuring_qmk/#change-vendor_id-product_id

**Review by fauxpark — APPROVED** [review]

File: keyboards/40percentclub/sixpack/sixpack.h
Comment (Erovia) [inline keyboards/40percentclub/sixpack/sixpack.h:28]:
```suggestion
#define LAYOUT_ortho_2x3( \
```
I'd consider using this macro name, so if/when an `ortho_2x3` community layout is created, adding support will be trivial.

**Review by Erovia — APPROVED** [review]
I left a small recommendation.
It's not a show-stopper, so I leave it up to you decide if you implement it or not.

File: keyboards/40percentclub/sixpack/config.h
Comment (drashna) [reply keyboards/40percentclub/sixpack/config.h:23]: I mean, I do understand that, but I didn't catch this on the other ones.

For instance, ZSA (ergodox ez, moonlander) was assigned 0x3297 by USB-IF last year.  And 0x3339 has already been assigned (and appears to be the last assigned).  That means that 0x3430 may/will be assigned in the very near future, hence my concern about this low of a number.

I would personally prefer 0xCEEB over 0x3430.

Comment (zvecr) [reply keyboards/40percentclub/sixpack/config.h:23]: Given it exists in the repo already, I think it best to align for now. And bulk change all the existing ones if we still see it as an issue.

File: keyboards/40percentclub/sixpack/keymaps/via/keymap.c
Comment (zvecr) [inline keyboards/40percentclub/sixpack/keymaps/via/keymap.c:38]:
```suggestion
    [0] = LAYOUT_ortho_2x3(KC_MUTE, KC_UP  , KC_MPLY,
		 KC_LEFT, KC_DOWN, KC_RGHT),
    [1] = LAYOUT_ortho_2x3(KC_TRNS, KC_TRNS, KC_TRNS,
		 KC_TRNS, KC_TRNS, KC_TRNS),
    [2] = LAYOUT_ortho_2x3(KC_TRNS, KC_TRNS, KC_TRNS,
		 KC_TRNS, KC_TRNS, KC_TRNS),
    [3] = LAYOUT_ortho_2x3(KC_TRNS, KC_TRNS, KC_TRNS,
		 KC_TRNS, KC_TRNS, KC_TRNS),
    [4] = LAYOUT_ortho_2x3(KC_TRNS, KC_TRNS, KC_TRNS,
		 KC_TRNS, KC_TRNS, KC_TRNS),
    [5] = LAYOUT_ortho_2x3(KC_TRNS, KC_TRNS, KC_TRNS,
```

File: keyboards/40percentclub/sixpack/keymaps/fkeys/keymap.c
Comment (zvecr) [inline keyboards/40percentclub/sixpack/keymaps/fkeys/keymap.c:46]:
```suggestion
    [0] = LAYOUT_ortho_2x3(/* F13 to F24 */
```

File: keyboards/40percentclub/sixpack/keymaps/default/keymap.c
Comment (zvecr) [inline keyboards/40percentclub/sixpack/keymaps/default/keymap.c:28]:
```suggestion
    [0] = LAYOUT_ortho_2x3(/* Default */
```

File: keyboards/40percentclub/sixpack/info.json
Comment (zvecr) [inline keyboards/40percentclub/sixpack/info.json:8]:
```suggestion
        "LAYOUT_ortho_2x3": {
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**
